src/utils/time.ts

```diff
@@ -10,9 +10,9 @@ export function sleep(durationInMillisecond): Promise<void> {
 }
 
 /**
- * returns system time as a plain ISO date string with microsecond precision
+ * returns an UTC ISO-8601 timestamp with microsecond precision
  * using @js-temporal/polyfill
  */
 export function getCurrentTimeInHighPrecision(): string {
-  return Temporal.Now.plainDateTimeISO().toString({ smallestUnit: 'microseconds' });
-}
\ No newline at end of file
+  return Temporal.Now.instant().toString({ smallestUnit: 'microseconds'  });
```

Comment:
would something like `(new Date()).toISOString()` work instead?

seems a bit unnecessary to load a polyfill for `Temporal` just to do something this basic 😕

Reply:
agreed. @thehenrytsai might have context here. was the reason behind using `Temporal` to get microsecond precision?

Reply:
also why not just use a timestamp instead of a string?  that's immediately more sortable and is significantly simpler to parse

Reply:
deferring to @thehenrytsai here for context

Reply:
@mistermoe  and @dcrousso it is indeed the microsecond precision requirement.

Reply:
ok that explains `Temporal`, but why not store the timestamp instead of a string?

Reply:
@dcrousso, what timestamp format are you suggesting?

Reply:
like a raw number

Reply:
how important is this timestamp w.r.t ordering @thehenrytsai ? is it relied upon in other parts of the SDK?

Reply:
the format shouldn't really matter for ordering (i.e. whether it's a string or a number is kinda irrelevant since theyre both just different ways of expressing the same underlying value)

also wouldn't a number be easier to compare?

Reply:
@dcrousso, what raw number though? It needs to represent a timestamp in with microsecond precision.
@mistermoe, it is how message are compared for a given so it's important. Though I personally don't care about precision, milli, micro, nano, whatever the DWN spec wants.

Reply:
maybe do it like JS does with `Date.now()` (which has millisecond precision) except represent the nanoseconds as a decimal (e.g. `1234567890123.456` would be the same as `"2009-02-13T23:31:30.123456Z"`)?  or maybe we just have it be a nanosecond precision value (e.g. `1234567890123456`)?

so long as it's clear in the documentation i think we can make this whatever format we want

Reply:
@dcrousso, my 1st preference is to use a well-known standard/convention and over coming up with our own. ISO 8601 seems reasonable to me still.

Reply:
Agreed to file issue to track: #230.